File: vscode-extension/src/ide/settings/agentSettingsModels.ts
/**
 * Agent Settings Models
 * 
 * Type definitions and enums for agent settings.
 * Translated from jetbrains-plugin/src/main/kotlin/com/asakii/settings/AgentSettingsModels.kt
 */

/**
 * Default thinking level enum
 * 
 * Simplified to three core levels: Off, Think, Ultra
 */
export enum DefaultThinkingLevel {
    OFF = 'OFF',
    THINK = 'THINK',
    ULTRA = 'ULTRA'
}

export const DefaultThinkingLevelInfo: Record<DefaultThinkingLevel, { displayName: string; description: string }> = {
    [DefaultThinkingLevel.OFF]: { displayName: 'Off', description: 'Disable extended thinking' },
    [DefaultThinkingLevel.THINK]: { displayName: 'Think', description: 'Standard thinking for most tasks' },
    [DefaultThinkingLevel.ULTRA]: { displayName: 'Ultra', description: 'Deep thinking for complex tasks' }
};

/**
 * Get DefaultThinkingLevel from string name
 */
export function getDefaultThinkingLevelFromName(name: string | null | undefined): DefaultThinkingLevel | undefined {
    if (!name) return undefined;
    const normalized = name.toUpperCase();
    return Object.values(DefaultThinkingLevel).find(level => level === normalized);
}

/**
 * Thinking level configuration
 * 
 * Used to store full thinking level info, including preset and custom levels
 */
export interface ThinkingLevelConfig {
    /** Unique identifier: off, think, ultra, custom_xxx */
    id: string;
    /** Display name */
    name: string;
    /** Token count */
    tokens: number;
    /** Whether it's a custom level */
    isCustom: boolean;
}

/**
 * Generic option configuration
 * 
 * Used for dropdown options, supports dynamic return to frontend
 */
export interface OptionConfig {
    /** Unique identifier */
    id: string;
    /** Display label */
    label: string;
    /** Description (optional) */
    description: string;
    /** Whether it's the default value */
    isDefault: boolean;
}

/**
 * Custom model configuration
 * 
 * Used to store user-defined model information
 */
export interface CustomModelConfig {
    /** Unique identifier (e.g., "custom_xxx") */
    id: string;
    /** Display name (e.g., "My Custom Model") */
    displayName: string;
    /** Model ID (e.g., "claude-sonnet-4-5-20250929") */
    modelId: string;
}

/**
 * External path rule type
 */
export enum ExternalPathRuleType {
    INCLUDE = 'INCLUDE',  // Allow access
    EXCLUDE = 'EXCLUDE'   // Deny access
}

/**
 * External path access rule
 * 
 * Used to control which paths outside the project can be accessed
 */
export interface ExternalPathRule {
    /** Directory path */
    path: string;
    /** Rule type */
    type: ExternalPathRuleType;
}

/**
 * Unified model information class
 * 
 * Used to uniformly represent built-in and custom models
 */
export interface ModelInfo {
    /** Actual model ID */
    modelId: string;
    /** Display name */
    displayName: string;
    /** Whether it's a built-in model */
    isBuiltIn: boolean;
}

/**
 * AI Agent Provider enum
 */
export enum AiAgentProvider {
    CLAUDE = 'CLAUDE',
    CODEX = 'CODEX'
}

/**
 * MCP Backend constants
 */
export const MCP_BACKEND_ALL = 'all';
export const MCP_BACKEND_CLAUDE = 'claude';
export const MCP_BACKEND_CODEX = 'codex';

/**
 * Agent configuration data class
 */
export interface AgentConfig {
    name: string;
    description: string;
    prompt: string;
    tools: string[];
    /** Main AI's sub-agent selection hint */
    selectionHint: string;
}

File: jetbrains-plugin/src/main/kotlin/com/asakii/settings/AgentSettingsModels.kt
package com.asakii.settings

import kotlinx.serialization.Serializable

/**
 * 默认思考等级枚举
 *
 * 简化为三个核心级别：Off、Think、Ultra
 */
enum class DefaultThinkingLevel(val displayName: String, val description: String) {
    OFF("Off", "Disable extended thinking"),
    THINK("Think", "Standard thinking for most tasks"),
    ULTRA("Ultra", "Deep thinking for complex tasks");

    companion object {
        fun fromName(name: String?): DefaultThinkingLevel? {
            return entries.find { it.name == name }
        }
    }
}

/**
 * 思考级别配置
 *
 * 用于存储思考级别的完整信息，包括预设级别和自定义级别
 */
@Serializable
data class ThinkingLevelConfig(
    val id: String,        // 唯一标识：off, think, ultra, custom_xxx
    val name: String,      // 显示名称
    val tokens: Int,       // token 数量
    val isCustom: Boolean = false  // 是否为自定义级别
)

/**
 * 通用选项配置
 *
 * 用于下拉列表选项，支持动态返回给前端
 */
@Serializable
data class OptionConfig(
    val id: String,           // 唯一标识
    val label: String,        // 显示名称
    val description: String = "",  // 描述（可选）
    val isDefault: Boolean = false // 是否为默认值
)

/**
 * 自定义模型配置
 *
 * 用于存储用户自定义的模型信息
 */
@Serializable
data class CustomModelConfig(
    val id: String,        // 唯一标识（如 "custom_xxx"）
    val displayName: String,  // 显示名称（如 "My Custom Model"）
    val modelId: String       // 模型 ID（如 "claude-sonnet-4-6"）
)

/**
 * 外部路径规则类型
 */
enum class ExternalPathRuleType {
    INCLUDE,  // 包含（允许访问）
    EXCLUDE   // 排除（禁止访问）
}

/**
 * 外部路径访问规则
 *
 * 用于控制哪些项目外部路径可以被访问
 */
@Serializable
data class ExternalPathRule(
    val path: String,                    // 目录路径
    val type: ExternalPathRuleType       // 规则类型
)

/**
 * 统一的模型信息类
 *
 * 用于统一表示内置模型和自定义模型
 */
data class ModelInfo(
    val modelId: String,      // 实际模型 ID
    val displayName: String,  // 显示名称
    val isBuiltIn: Boolean    // 是否为内置模型
)
